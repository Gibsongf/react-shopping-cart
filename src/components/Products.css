.container{
    display: grid;
    grid-template-columns: repeat(auto-fit,minmax(400px,2fr));
    justify-items: center;
    gap: 50px;
}