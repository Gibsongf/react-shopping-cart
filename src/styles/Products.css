.container{
    display: flex;
    justify-items: center;
    flex-wrap: wrap;
    justify-content: center;
    
}
.item {
    margin: 25px;
}