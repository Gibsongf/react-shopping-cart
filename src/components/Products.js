import "../styles/Products.css";
import CurrencyIcon from "../imgs/Gold_Currency_Icon.png";
import { NavLink } from "react-router-dom";
export const Product = (props) => {
    const { name, imgSrc, price } = props;

    const handleClick = () => {
        // props.clickedItem({ ...props });
        localStorage.setItem("selected-product", JSON.stringify(props));
    };

    return (
        <NavLink
            className="item"
            id="item"
            onClick={handleClick}
            to={`${props.id}`}
            name="item"
        >
            <img src={imgSrc} alt={name} className="item-img" />
            <div className="price-name">
                <h4>{name}</h4>
                <img className="gold-icon" src={CurrencyIcon} alt="gold-coin" />
                {price}
            </div>
        </NavLink>
    );
};
