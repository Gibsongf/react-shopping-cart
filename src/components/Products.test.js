import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import { Product } from "./Products";
import Src from "../imgs/tactics/Alchemic_Carbine_Icon.png";
const ProductSetup = () => {
	const name = "The Alchemic Carbine";
	const description =
		"Crossbow-type ranged weapon which poisons enemies hit by its bolts.";
	const price = 2000;
	return (
		<Product
			name={name}
			description={description}
			price={price}
			imgSrc={Src}
		/>
	);
};

describe("open and close window of the product", () => {
	it('without item click content',()=>{
		const { container } = render(<ProductSetup />);
    	expect(container).toMatchSnapshot();
	})
	it("pop up on click", () => {
		const { container } = render(<ProductSetup />);
		const img = screen.getByRole("img", { name: "The Alchemic Carbine" });
		fireEvent.click(img);
		expect(!container).toMatchSnapshot();
	});
	it("close on click", () => {
		const { container } = render(<ProductSetup />);
		const img = screen.getByRole("img", { name: "The Alchemic Carbine" });
		fireEvent.click(img);
		const btnClose = screen.getByRole("button", { name: "close-icon" });
		fireEvent.click(btnClose);
		expect(container).toMatchSnapshot();

	});
});
