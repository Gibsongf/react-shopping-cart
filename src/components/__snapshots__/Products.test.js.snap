// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open and close window of the product close on click 1`] = `
<div>
  <div
    class="item"
    name="item"
  >
    <img
      alt="The Alchemic Carbine"
      class="item-img"
      src="Alchemic_Carbine_Icon.png"
    />
    <div
      class="price-name"
    >
      <h4>
        The Alchemic Carbine
      </h4>
      <img
        alt="gold-coin"
        class="gold-icon"
        src="Gold_Currency_Icon.png"
      />
      2000
    </div>
  </div>
</div>
`;

exports[`open and close window of the product pop up on click 1`] = `false`;

exports[`open and close window of the product without item click content 1`] = `
<div>
  <div
    class="item"
    name="item"
  >
    <img
      alt="The Alchemic Carbine"
      class="item-img"
      src="Alchemic_Carbine_Icon.png"
    />
    <div
      class="price-name"
    >
      <h4>
        The Alchemic Carbine
      </h4>
      <img
        alt="gold-coin"
        class="gold-icon"
        src="Gold_Currency_Icon.png"
      />
      2000
    </div>
  </div>
</div>
`;
